platforms/software/software-diagnostics/src/main/java/org/gradle/api/plugins/SoftwareReportingTasksPlugin.java

```diff
@@ -0,0 +1,164 @@
+import org.gradle.api.tasks.TaskContainer;
+import org.gradle.api.tasks.diagnostics.ArtifactTransformsReportTask;
+import org.gradle.api.tasks.diagnostics.BuildEnvironmentReportTask;
+import org.gradle.api.tasks.diagnostics.DependencyInsightReportTask;
+import org.gradle.api.tasks.diagnostics.DependencyReportTask;
+import org.gradle.api.tasks.diagnostics.OutgoingVariantsReportTask;
+import org.gradle.api.tasks.diagnostics.ResolvableConfigurationsReportTask;
+import org.gradle.api.tasks.diagnostics.TaskReportTask;
+import org.gradle.api.tasks.diagnostics.DiagnosticsTaskNames;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Adds various reporting tasks that provide information about a software project, such as information about
+ * configurations and dependency reporting tasks that make use of dependency management.
+ *
+ * @since 8.13
+ */
+@Incubating
+@Nonnull
+public abstract class SoftwareReportingTasksPlugin implements Plugin<Project>, DiagnosticsTaskNames {
+    @Override
+    public void apply(Project project) {
+        final TaskContainer tasks = project.getTasks();
+
+        // static classes are used for the actions to avoid implicitly dragging project/tasks into the model registry
+        String projectName = project.toString();
+
+        tasks.register(HelpTasksPlugin.DEPENDENCY_INSIGHT_TASK, DependencyInsightReportTask.class, new DependencyInsightReportTaskAction(projectName));
+        tasks.register(HelpTasksPlugin.DEPENDENCIES_TASK, DependencyReportTask.class, new DependencyReportTaskAction(projectName));
+        tasks.register(BuildEnvironmentReportTask.TASK_NAME, BuildEnvironmentReportTask.class, new BuildEnvironmentReportTaskAction(projectName));
+        registerDeprecatedSoftwareModelTasks(tasks, projectName);
+        tasks.register(HelpTasksPlugin.OUTGOING_VARIANTS_TASK, OutgoingVariantsReportTask.class, task -> {
+            task.setDescription("Displays the outgoing variants of " + projectName + ".");
+            task.setGroup(HelpTasksPlugin.HELP_GROUP);
+            task.setImpliesSubProjects(true);
+            task.getShowAll().convention(false);
+        });
+        tasks.register(HelpTasksPlugin.RESOLVABLE_CONFIGURATIONS_TASK, ResolvableConfigurationsReportTask.class, task -> {
+            task.setDescription("Displays the configurations that can be resolved in " + projectName + ".");
+            task.setGroup(HelpTasksPlugin.HELP_GROUP);
+            task.setImpliesSubProjects(true);
+            task.getShowAll().convention(false);
+            task.getRecursive().convention(false);
+        });
+        tasks.register(HelpTasksPlugin.ARTIFACT_TRANSFORMS_TASK, ArtifactTransformsReportTask.class, task -> {
+            task.setDescription("Displays the Artifact Transforms that can be executed in " + projectName + ".");
+            task.setGroup(HelpTasksPlugin.HELP_GROUP);
+            task.setImpliesSubProjects(true);
+        });
+        tasks.withType(TaskReportTask.class).configureEach(task -> {
```

Comment:
This seems like it should be part of `HelpTasksPlugin`

Reply:
Agreed, will update in a followup.